### PR TITLE
Fix the instantiate command for Substrate 0.9.42-based chains

### LINKED ISSUE #1563: Failure when interacting with chains that use older `pallet-contracts` versions
I'm trying to instantiate `flipper` from `ink-examples` on either Pendulum or Ternoa.

`cargo-contract` fails with an error due to issues with metadata. I believe those chains use an older version of `pallet-contracts`.

```
➜  flipper git:(heads/v5.x.x) ✗ cargo contract instantiate --url wss://pendulum-rpc.dwellir.com --args true --suri //Alice -x
Warning: This chain does not yet support checking for compatibility of your contract types (https://use.ink/faq#type-comparison).
 Dry-running new (skip with --skip-dry-run)
ERROR: Could not decode `ContractResult::events`:
	Could not decode variant byte for `Option`:
		Not enough data to fill buffer


Caused by:
    0: Could not decode variant byte for `Option`:
       	Not enough data to fill buffer
       
    1: Not enough data to fill buffer


➜  flipper git:(heads/v5.x.x) ✗ cargo contract instantiate --url wss://mainnet.ternoa.network --args true --suri //Alice -x
Warning: This chain does not yet support checking for compatibility of your contract types (https://use.ink/faq#type-comparison).
 Dry-running new (skip with --skip-dry-run)
ERROR: Could not decode `ContractResult::events`:
	Could not decode variant byte for `Option`:
		Not enough data to fill buffer


Caused by:
    0: Could not decode variant byte for `Option`:
       	Not enough data to fill buffer
       
    1: Not enough data to fill buffer
```

### PR DESCRIPTION
**Pull request by smiasojed**
## Summary
Closes #1563
- [ ] y/n | Does it introduce breaking changes?
- [ ] y/n | Is it dependent on the specific version of `ink` or `pallet-contracts`?

## Description
Fix instantiate command
Add `cargo contract storage --version` command

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [x] I have added an entry to `CHANGELOG.md`
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: crates/extrinsics/src/pallet_contracts_primitives.rs [hunk 1]
```diff
@@ -78,21 +78,15 @@ pub struct ContractResult<R, Balance, EventRecord> {
     pub debug_message: Vec<u8>,
     /// The execution result of the wasm code.
     pub result: R,
-    /// The events that were emitted during execution. It is an option as event
-    /// collection is optional.
-    pub events: Option<Vec<EventRecord>>,
```
Comment (cmichi) [inline crates/extrinsics/src/pallet_contracts_primitives.rs:83]: What about chains that still have the `events` field?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+- Fix the `instantiate` command for Substrate `0.9.42` based chains - [#1546](https://github.com/paritytech/cargo-contract/pull/1564)
+
 ### Added
```
Comment (cmichi) [inline CHANGELOG.md:12]: - Add `cargo contract storage --version` command - [#1546](https://github.com/paritytech/cargo-contract/pull/1564)

File: crates/extrinsics/src/pallet_contracts_primitives.rs (hunk 1 above)
Comment (smiasojed) [reply crates/extrinsics/src/pallet_contracts_primitives.rs:83]: we do not use it in our code, so it is just not parsed from binary data.

Comment (cmichi) [reply crates/extrinsics/src/pallet_contracts_primitives.rs:83]: Where do the events depicted after calling/instantiating/etc. come from if not the `ContractResult`?

File: crates/cargo-contract/src/cmd/storage.rs [hunk 3]
```diff
@@ -90,55 +98,66 @@ impl StorageCommand {
...
-                }
-            }
-            Err(_) => {
-                eprintln!(
-                    "{} Displaying raw storage: no valid contract metadata artifacts found",
-                    "Info:".cyan().bold(),
-                );
+        if self.version {
+            println!("{}", storage_layout.version().await?);
+            Ok(())
```
Comment (cmichi) [inline crates/cargo-contract/src/cmd/storage.rs:103]: Please return early here to save us from the else indentation.

File: crates/cargo-contract/src/cmd/storage.rs [hunk 4]
```diff
@@ -65,6 +70,9 @@ pub struct StorageCommand {
     /// Path to the `Cargo.toml` of the contract.
     #[clap(long, value_parser)]
     manifest_path: Option<PathBuf>,
+    /// Fetch the storage version of the pallet contracts (pallet version).
```
Comment (cmichi) [inline crates/cargo-contract/src/cmd/storage.rs:73]:
```suggestion
    /// Fetch the storage version of the pallet contracts (state query: contracts::palletVersion()).
```

I think this makes it clearer what is being fetched.

**Review by cmichi — APPROVED** [review]

**Pull request merged**